app/screens/edit_profile/edit_profile.tsx

```diff
@@ -0,0 +1,399 @@
+import Field from './components/field';
+import ProfileError from './components/profile_error';
+
+const edges: Edge[] = ['bottom', 'left', 'right'];
+
+const styles = StyleSheet.create({
+    flex: {
+        flex: 1,
+    },
+    top: {
+        padding: 25,
+        alignItems: 'center',
+        justifyContent: 'center',
+    },
+    separator: {
+        height: 15,
+    },
+    footer: {
+        height: 40,
+        width: '100%',
+    },
+    spinner: {
+        position: 'absolute',
+        width: '100%',
+        height: '100%',
+        alignItems: 'center',
+        justifyContent: 'center',
+        zIndex: 1000,
+    },
+});
+
+const SPINNER_LAYER = 'Shape Layer 1';
+
+const EditProfile = ({
+    closeButtonId,
+    componentId,
+    currentUser,
+    isModal,
+    isTablet,
+    lockedFirstName,
+    lockedLastName,
+    lockedNickname,
+    lockedPosition,
+}: EditProfileProps) => {
+    const intl = useIntl();
+    const serverUrl = useServerUrl();
+    const theme = useTheme();
+    const keyboardAwareRef = useRef<KeyboardAwareScrollView>();
+
+    const firstNameRef = useRef<FloatingTextInputRef>(null);
+    const lastNameRef = useRef<FloatingTextInputRef>(null);
+    const usernameRef = useRef<FloatingTextInputRef>(null);
+    const emailRef = useRef<FloatingTextInputRef>(null);
+    const nicknameRef = useRef<FloatingTextInputRef>(null);
+    const positionRef = useRef<FloatingTextInputRef>(null);
+
+    const [userInfo, setUserInfo] = useState<UserInfo>({
+        email: currentUser.email,
+        firstName: currentUser.firstName,
+        lastName: currentUser.lastName,
+        nickname: currentUser.nickname,
+        position: currentUser.position,
+        username: currentUser.username,
+    });
+    const [canSave, setCanSave] = useState(false);
+    const [error, setError] = useState<ErrorText | undefined>();
+    const [updating, setUpdating] = useState(false);
+
+    const scrollViewRef = useRef<KeyboardAwareScrollView>();
+
+    const buttonText = intl.formatMessage({id: 'mobile.account.settings.save', defaultMessage: 'Save'});
+    const rightButton = isTablet ? null : {
+        id: 'update-profile',
+        enabled: false,
+        showAsAction: 'always' as const,
+        testID: 'edit_profile.save.button',
+        color: theme.sidebarHeaderTextColor,
+        text: buttonText,
+    };
+
+    useEffect(() => {
+        const unsubscribe = Navigation.events().registerComponentListener({
+            navigationButtonPressed: ({buttonId}: { buttonId: string }) => {
+                switch (buttonId) {
+                    case 'update-profile':
+                        submitUser();
+                        break;
+                    case closeButtonId:
+                        close();
+                        break;
+                }
+            },
+        }, componentId);
+
+        return () => {
+            unsubscribe.remove();
+        };
+    }, [userInfo, isTablet]);
```

Comment:
I get a bit paranoid with callbacks and effects, so bear with me.

Technically, this effect depends on `submitUser`, `close`, and `componentId`.
I understand that we do not consider `componentId` since it should not change during the lifetime of the component.

`submitUser` gets defined on every render (it is not created using `useCallback`). In order to deal with that, we could (as you did) add here the stuff that actually change the behaviour of `submitUser`. As I can see, the things that `submitUser` depends on are:
- `enableSaveButton` (which depends on `componentId` and `rightButton`).
- `userInfo` (already in the dependencies)
- `severUrl` (should not change during the lifetime of the object)
- `resetScreen` (but this one does not depends on anything)

With close, something similar happens, and the dependencies are:
- `isModal`
- `isTablet` (already in the dependencies)
- `componentId`

The only final piece to analyze is the right button, which depends on:
- `isTablet`
- `intl`
- `theme`

This leave us with the following stuff not added to the dependency of the effect:
- `intl`
- `theme`
- `componentId`
- `severUrl`
- `isModal`

Can we safely assume all of them will not change during the lifetime of the object? I think that at least for theme we cannot.